pkg/macos/pkg-scripts/preinstall

```diff
@@ -96,12 +96,20 @@ if [ -L "$SBIN_DIR/salt-config" ]; then
 fi
 
 #-------------------------------------------------------------------------------
-# Remove the $INSTALL_DIR directory
+# Remove folders and files from the INSTALL_DIR
+# Don't remove extras-3.10
 #-------------------------------------------------------------------------------
-if [ -d "$INSTALL_DIR" ]; then
-    log "Cleanup: Removing $INSTALL_DIR"
-    rm -rf "$INSTALL_DIR"
-    log "Cleanup: Removed Successfully"
+for dir in "$INSTALL_DIR"/*/; do
+    if [[ "$dir" != *"extras-3."* ]]; then
+        log "Cleanup: Removing $dir"
+        rm -rf "$dir"
+    fi
+done
+log "Cleanup: Removed Directories Successfully"
+
+if [ -f "$INSTALL_DIR/salt-minion" ]; then
+    find $INSTALL_DIR -maxdepth 1 -type f -delete
+    log "Cleanup: Removed Files Successfully"
```

Comment:
So, if `$INSTALL_DIR/salt-minion` exists, then `extras-3.*` does not exist?
Because this call would also nuke that.

Reply:
ah, I missed the `-maxdepth 1` although I did see the `-type f` ... :/